.github/workflows/python-integration-tests.yml

```diff
@@ -0,0 +1,58 @@
+#
+# This workflow will run all python integrations tests.
+#
+
+name: python-integration-tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main"]
+  pull_request:
+    branches: [ "main"] # TODO: remove before pushing to main
+  schedule:
+    - cron: '0 */12 * * *' # Run every 12 hours: midnight UTC and noon UTC
+
+
+permissions:
+  contents: read
+
+jobs:
+  integration-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
```

Comment:
could you add to the PR a quick estimate about the number of AI requests the (current) tests will generate per day, considering the permutations? I'm sure is few cents, but just to have an idea

Reply:
done